frontend/src/metabase/writeback/components/ActionCreator/FormCreator/FieldSettingsPopover.tsx

```diff
@@ -0,0 +1,114 @@
+import React from "react";
+import { t } from "ttag";
+
+import TippyPopoverWithTrigger from "metabase/components/PopoverWithTrigger/TippyPopoverWithTrigger";
+import type {
+  FieldSettings,
+  FieldType,
+  InputType,
+} from "metabase-types/api/writeback";
+
+import Radio from "metabase/core/components/Radio";
+import Icon from "metabase/components/Icon";
+
+import { fieldTypes, inputTypes } from "./constants";
+import {
+  SettingsPopoverBody,
+  SectionLabel,
+  FieldTypeWrapper,
+} from "./FieldSettingsPopover.styled";
+
+export function FieldSettingsPopover({
+  fieldSettings,
+  onChange,
+}: {
+  fieldSettings: FieldSettings;
+  onChange: (fieldSettings: any) => void;
+}) {
+  return (
+    <TippyPopoverWithTrigger
+      placement="bottom-end"
+      triggerContent={<Icon name="gear" size={16} />}
+      maxWidth={400}
+      popoverContent={() => (
+        <FormCreatorPopoverBody
+          fieldSettings={fieldSettings}
+          onChange={onChange}
+        />
+      )}
+    />
+  );
+}
+
+export function FormCreatorPopoverBody({
+  fieldSettings,
+  onChange,
+}: {
+  fieldSettings: FieldSettings;
+  onChange: (fieldSettings: FieldSettings) => void;
+}) {
+  const updateFieldType = (newFieldType: FieldType) =>
+    onChange({
+      ...fieldSettings,
+      fieldType: newFieldType,
+    });
+
+  const updateInputType = (newInputType: InputType) =>
+    onChange({
+      ...fieldSettings,
+      inputType: newInputType,
+    });
```

Comment:
super nit: guess the preferred naming convention for such callbacks _inside_ the component here is `handle*` like `handleFieldTypeChange` and `handleInputTypeChange`.

As a side note, when you pass a callback to a child component or receive a callback from a parent, think `on*` names are preferred (like `onChange` and `onBlur`)